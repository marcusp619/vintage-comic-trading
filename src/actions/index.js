export const isLoading = bool => ({
  type: 'IS_LOADING',
  isLoading: bool,
});

export const hasErrored = bool => ({
  type: 'HAS_ERRORED',
  hasErrored: bool,
});

export const comicFetchDataSuccess = comic => ({
  type: 'COMICS_FETCH_DATA_SUCCESS',
  comic,
});

export const addComicToUser = comic => ({
  type: 'ADD_COMIC_TO_USER',
  comic,
});

export const userSignedIn = user => ({
  type: 'USER_SIGNED_IN',
  user,
});

export const userSignedOut = user => ({
  type: 'USER_SIGNED_OUT',
  user,
});
