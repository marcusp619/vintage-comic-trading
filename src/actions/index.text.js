import * as actions from './'

describe('actions', () => {
  it('should have a type of IS_LOADING', () => {
    expect(true).toEqual(false);
  })
})
