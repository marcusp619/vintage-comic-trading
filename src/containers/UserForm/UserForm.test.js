import React from "react";
import { shallow, mount } from "enzyme";
import {
  UserForm,
  mapStateToProps,
  mapDispatchToProps,
  Input,
  Form
} from "./index.js";
import { userSignedIn } from "../../actions";

describe("UserForm Container", () => {
  it("should match snapshot", () => {
    //setup
    const wrapper = shallow(<UserForm />);

    //expectation
    expect(wrapper).toMatchSnapshot();
  });

  it("should set state with the correct keys and values", () => {
    const wrapper = shallow(<UserForm />);
    const userInput = wrapper.find(Input);

    userInput.simulate("change", {
      target: { user: "user", value: "Mark" }
    });

    expect(wrapper.state()).toEqual({ user: "Mark" });
  });

  describe("handleFormSubmit function", () => {
    let mockForm;
    let wrapper;
    let mockGetUser;
    let mockToggleSignInForm;
    beforeEach(() => {
      mockGetUser = jest.fn();
      mockToggleSignInForm = jest.fn();
      wrapper = shallow(
        <UserForm
          getUser={mockGetUser}
          toggleSignInForm={mockToggleSignInForm}
        />
      );
      mockForm = wrapper.find(Form);
    });

    it("should call handleFormSubmit with the correct params", () => {
      mockForm.simulate("submit", { preventDefault() {} });
      expect(mockGetUser).toHaveBeenCalled();
      expect(mockToggleSignInForm).toHaveBeenCalled();
    });
  });

  describe("mapstateToProps", () => {
    it("mapStateToProps", () => {
      //setup
      const mockState = {
        user: "mark"
      };
      const expected = {
        user: "mark"
      };

      //execution
      const mappedProps = mapStateToProps(mockState);

      //expectation
      expect(mappedProps).toEqual(expected);
    });
  });

  describe("mapDispatchToProps", () => {
    it("calls dispatch with an userSignedIn action", () => {
      //setup
      const mockDispatch = jest.fn();
      const actionToDispatch = userSignedIn("mark");

      //execution
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getUser("mark");

      //expectation
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
