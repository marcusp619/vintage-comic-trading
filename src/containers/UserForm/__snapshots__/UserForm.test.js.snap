// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm Container should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <label>
          UserName:
          <ForwardRef
            name="user"
            onChange={[Function]}
            value=""
          />
        </label>,
        <ForwardRef>
          Submit
        </ForwardRef>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "UserName:",
            <ForwardRef
              name="user"
              onChange={[Function]}
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "UserName:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "user",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "rules": Array [
                  "
  display: block;
  width: 500px;
  height: 2em;
",
                ],
              },
              "displayName": "styled.input",
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "input",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  background: transparent;
  border-radius: 3px;
  border: 2px solid #AF7A6D;
  color: #AF7A6D;
  display: block;
  font-size: 1em;
  margin 0 1em;
  padding: 0.45em 1.25em;
  transition: all .75s ease-out;

  &:hover {
    background: #AF7A6D;
    color: #fff;
  }
",
            ],
          },
          "displayName": "styled.button",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  background: rgba(226, 212, 186, 0.8);
  color: #af7a6d;
  display: grid;
  justify-content: center;
  font-size: 1.2em;
  font-weight: bold;
  align-items: center;
  height: 400px;
  position: absolute;
  top: 180%;
  left: 20%;
  width: 50em;
  z-index: 2;
",
        ],
      },
      "displayName": "styled.form",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "form",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <label>
            UserName:
            <ForwardRef
              name="user"
              onChange={[Function]}
              value=""
            />
          </label>,
          <ForwardRef>
            Submit
          </ForwardRef>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "UserName:",
              <ForwardRef
                name="user"
                onChange={[Function]}
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "UserName:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "user",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "rules": Array [
                    "
  display: block;
  width: 500px;
  height: 2em;
",
                  ],
                },
                "displayName": "styled.input",
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  background: transparent;
  border-radius: 3px;
  border: 2px solid #AF7A6D;
  color: #AF7A6D;
  display: block;
  font-size: 1em;
  margin 0 1em;
  padding: 0.45em 1.25em;
  transition: all .75s ease-out;

  &:hover {
    background: #AF7A6D;
    color: #fff;
  }
",
              ],
            },
            "displayName": "styled.button",
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  background: rgba(226, 212, 186, 0.8);
  color: #af7a6d;
  display: grid;
  justify-content: center;
  font-size: 1.2em;
  font-weight: bold;
  align-items: center;
  height: 400px;
  position: absolute;
  top: 180%;
  left: 20%;
  width: 50em;
  z-index: 2;
",
          ],
        },
        "displayName": "styled.form",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "form",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
